File: app/src/main/java/com/example/despensacx/ListaAdapter.java
package com.example.despensacx;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class ListaAdapter extends RecyclerView.Adapter<ListaAdapter.ListaViewHolder> {

    public interface OnListaClickListener {
        void onClick(ListaEntity lista);
        void onLongClick(ListaEntity lista, View view);
        void onEditar(ListaEntity lista);
        void onDuplicar(ListaEntity lista);
        void onEliminar(ListaEntity lista);
    }

    private List<ListaEntity> listas = new ArrayList<>();
    private final OnListaClickListener listener;

    public ListaAdapter(OnListaClickListener listener) {
        this.listener = listener;
    }

    public void setListas(List<ListaEntity> listas) {
        this.listas = listas;
        notifyDataSetChanged();
    }

    public ListaEntity getListaAt(int position) {
        return listas.get(position);
    }

    @NonNull
    @Override
    public ListaViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lista, parent, false);
        return new ListaViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ListaViewHolder holder, int position) {
        ListaEntity lista = listas.get(position);
        holder.bind(lista, listener);
    }

    @Override
    public int getItemCount() {
        return listas.size();
    }

    static class ListaViewHolder extends RecyclerView.ViewHolder {
        TextView tvNombre, tvFechaCreacion, tvUltimaModificacion, tvPresupuesto;
        ImageButton btnEditar, btnDuplicar, btnEliminar;

        public ListaViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNombre = itemView.findViewById(R.id.tvNombreLista);
            tvFechaCreacion = itemView.findViewById(R.id.tvFechaCreacion);
            tvUltimaModificacion = itemView.findViewById(R.id.tvUltimaModificacion);
            tvPresupuesto = itemView.findViewById(R.id.tvPresupuesto);
            btnEditar = itemView.findViewById(R.id.btnEditarLista);
            btnDuplicar = itemView.findViewById(R.id.btnDuplicarLista);
            btnEliminar = itemView.findViewById(R.id.btnEliminarLista);
        }

        public void bind(ListaEntity lista, OnListaClickListener listener) {
            String title = lista.getNombre() + (lista.isArchivada() ? " (Archivada)" : "");
            tvNombre.setText(title);
            tvFechaCreacion.setText("Creado: " + lista.getFechaCreacion());
            tvUltimaModificacion.setText("Modificado: " + lista.getFechaModificacion());

            if (lista.getPresupuestoMaximo() > 0) {
                tvPresupuesto.setVisibility(View.VISIBLE);
                tvPresupuesto.setText(String.format(Locale.getDefault(), "Presupuesto: $%.2f MXN", lista.getPresupuestoMaximo()));
            } else {
                tvPresupuesto.setVisibility(View.GONE);
            }

            itemView.setOnClickListener(v -> listener.onClick(lista));
            itemView.setOnLongClickListener(v -> {
                listener.onLongClick(lista, v);
                return true;
            });

            btnEditar.setOnClickListener(v -> listener.onEditar(lista));
            btnDuplicar.setOnClickListener(v -> listener.onDuplicar(lista));
            btnEliminar.setOnClickListener(v -> listener.onEliminar(lista));
        }
    }
}

File: app/src/main/kotlin/com/example/despensacx/adapter/ListaAdapter.kt
package com.example.despensacx.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.despensacx.R
import com.example.despensacx.data.ListaEntity
import java.util.*

class ListaAdapter(private val listener: OnListaClickListener) :
    RecyclerView.Adapter<ListaAdapter.ListaViewHolder>() {

    interface OnListaClickListener {
        fun onClick(lista: ListaEntity)
        fun onLongClick(lista: ListaEntity, view: View)
        fun onEditar(lista: ListaEntity)
        fun onDuplicar(lista: ListaEntity)
        fun onEliminar(lista: ListaEntity)
    }

    private var listas: List<ListaEntity> = emptyList()

    fun setListas(listas: List<ListaEntity>) {
        this.listas = listas
        notifyDataSetChanged()
    }

    fun getListaAt(position: Int): ListaEntity = listas[position]

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListaViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_lista, parent, false)
        return ListaViewHolder(view)
    }

    override fun onBindViewHolder(holder: ListaViewHolder, position: Int) {
        val lista = listas[position]
        holder.bind(lista, listener)
    }

    override fun getItemCount(): Int = listas.size

    class ListaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvNombre: TextView = itemView.findViewById(R.id.tvNombreLista)
        private val tvFechaCreacion: TextView = itemView.findViewById(R.id.tvFechaCreacion)
        private val tvUltimaModificacion: TextView = itemView.findViewById(R.id.tvUltimaModificacion)
        private val tvPresupuesto: TextView = itemView.findViewById(R.id.tvPresupuesto)
        private val btnEditar: ImageButton = itemView.findViewById(R.id.btnEditarLista)
        private val btnDuplicar: ImageButton = itemView.findViewById(R.id.btnDuplicarLista)
        private val btnEliminar: ImageButton = itemView.findViewById(R.id.btnEliminarLista)

        fun bind(lista: ListaEntity, listener: OnListaClickListener) {
            val title = lista.nombre + if (lista.archivada) itemView.context.getString(R.string.archivada_suffix) else ""
            tvNombre.text = title
            tvFechaCreacion.text = itemView.context.getString(R.string.creado_con_fecha, lista.fechaCreacion)
            tvUltimaModificacion.text = "Modificado: ${lista.fechaModificacion}"

            if (lista.presupuestoMaximo > 0) {
                tvPresupuesto.visibility = View.VISIBLE
                tvPresupuesto.text = String.format(Locale.getDefault(), "Presupuesto: $%,.2f MXN", lista.presupuestoMaximo)
            } else {
                tvPresupuesto.visibility = View.GONE
            }

            itemView.setOnClickListener { listener.onClick(lista) }
            itemView.setOnLongClickListener {
                listener.onLongClick(lista, it)
                true
            }

            btnEditar.setOnClickListener { listener.onEditar(lista) }
            btnDuplicar.setOnClickListener { listener.onDuplicar(lista) }
            btnEliminar.setOnClickListener { listener.onEliminar(lista) }
        }
    }
}
